web/src/narrow_state.ts

```diff
@@ -32,7 +32,11 @@ export function search_terms(current_filter: Filter | undefined = filter()): Nar
 }
 
 export function is_message_feed_visible(): boolean {
-    return !recent_view_util.is_visible() && !inbox_util.is_visible();
+    // It's important that `message_lists.current` is the
+    // source of truth for this since during the initial app load,
+    // `message_lists.current` is `undefined` and we don't want
+    // to return `true` if we haven't loaded the message feed yet.
+    return message_lists.current !== undefined;
 }
```

Comment:
This one could also be pulled out as a prep series; though I think removing the `inbox_util/recent_view_util` imports should happen in this commit.